revno: 680 [merge]
author: Joshua Harlow <harlowja@yahoo-inc.com>
committer: Scott Moser <smoser@ubuntu.com>
branch nick: trunk.fix
timestamp: Fri 2012-10-05 09:39:53 -0400
Bug: https://bugs.launchpad.net/cloud-init/+bug/1061964
message:
  config-drive: map hostname to local-hostname
  
  Ensure that for config drive that we map 'hostname' to 'local-hostname' so that
  the modules work correctly with the cfgdrive style of data.
Index: quantal/cloudinit/sources/DataSourceConfigDrive.py
===================================================================
--- quantal.orig/cloudinit/sources/DataSourceConfigDrive.py	2012-08-24 17:23:35.413455000 -0400
+++ quantal/cloudinit/sources/DataSourceConfigDrive.py	2012-10-05 11:36:40.121800086 -0400
@@ -85,6 +85,16 @@
         md = results['metadata']
         md = util.mergedict(md, DEFAULT_METADATA)
 
+        # Perform some metadata 'fixups'
+        #
+        # OpenStack uses the 'hostname' key
+        # while most of cloud-init uses the metadata
+        # 'local-hostname' key instead so if it doesn't
+        # exist we need to make sure its copied over.
+        for (tgt, src) in [('local-hostname', 'hostname')]:
+            if tgt not in md and src in md:
+                md[tgt] = md[src]
+
         user_dsmode = results.get('dsmode', None)
         if user_dsmode not in VALID_DSMODES + (None,):
             LOG.warn("user specified invalid mode: %s" % user_dsmode)
